lib/puppet/transaction.rb

```diff
@@ -287,15 +291,21 @@ def failed_dependencies?(resource)
     # explosion of edges, we also ended up reporting failures for containers
     # like class and stage.  This is undesirable; while just skipping the
     # output isn't perfect, it is RC-safe. --daniel 2011-06-07
-    suppress_report = (resource.class == Puppet::Type.type(:whit))
-
+    class_or_stage = resource.class == Puppet::Type.type(:whit)
```

Comment:
`class_or_stage` will be false if the resource.class is a stage, so maybe call it `is_class`?

Reply:
Yes, but `is_puppet_class` to be more specific.